src/odemis/driver/technolution.py

```diff
@@ -0,0 +1,544 @@
+                logging.info("Cleared the remainder of the acquisition queue upon termination")
+                return
+
+        clear_queue(self.acq_queue.qsize())
+        self.acq_queue.put(("terminate", None))
+
+    def convert_field_num2pixels(self, field_num, mega_field_data):
+        mega_field_data = self._megafield_metadata
+        return (field_num[0] * self._shape.value[0] * mega_field_data.x_eff_cell_size,
+                field_num[1] * self._shape.value[1] * mega_field_data.y_eff_cell_size)
+
+    def _getMetadata(self):
+        """
+        Create dict containing all metadata from siblings and own metadata
+        """
+        md = {}
+        for md_dev in self.md_devices:
+            for key in md_dev.keys():
+                if key not in md:
+                    md[key] = md_dev[key]
+                elif key in (model.MD_HW_NAME, model.MD_HW_VERSION, model.MD_SW_VERSION):
+                    md[key] = ", ".join([md[key], md_dev[key]])
+        return md
+
+    def _data_content2value(self, content):
+        """
+        Converts string with options for the representation of an image into a value for an ASM call
+        :param content:  string holding one of the accepted options
+        :return:
+        """
+        if content == 'thumbnail':
+            logging.error("Option %s not yet implemented" % content)
+            return True
+        elif content == 'full':
+            logging.error("Option %s not yet implemented" % content)
+            return False
+        else:
+            # logging.error("Option %s not yet implemented" % content)
+            raise ValueError("Option %s not yet implemented" % content)
+
+    def _setAcqDelay(self, input_delay):
+        if input_delay - self.parent._EBeamScanner.scanDelay.value[0] >= 0:
+            return input_delay
+        else:
+            logging.error("Wrong input x_scan_to_acq_delay got a negative value.\n"
+                          "Change values so that 'self.acqDelay.value - self.parent._EBeamScanner.scanDelay.value[0]' has a positive result")
+            return self.acqDelay.value
+
+class ASMDataFlow(model.DataFlow):
+    def __init__(self, detector, start_func, next_func, stop_func, get_single_field_func):
+        super(ASMDataFlow, self).__init__(self)
+
+        self._start = start_func
+        self._next = next_func
+        self._stop = stop_func
+        self._single_field = get_single_field_func
+
+        self.active = False
+
+        self._acquisition_lock = threading.Lock()
+
+    def start_generate(self):
+        """
+        Start the dataflow.
+        """
+        with self._acquisition_lock:
+            self._start()
+            self.active = True
+
+    def next(self, field_num):
+        with self._acquisition_lock:
+            self._next(field_num)
+
+    def stop_generate(self):
+        """
+        Stop the dataflow.
+        """
+        self._stop()
+        self.active = False
+
+    def get(self, asap=True):
```

Comment:
That's part of the get() API. It means that as soon as an image is returned, it'll pass it. Otherwise, it'll make sure to wait for an image which started after the call to .get() started. It's only meaningful if the acquisition is already going on (because another client has called `.subscribe()`). So here, as it's not allowed to call the function while an acquisition is already active, this is not meaningful (because it'll always behave as if `asap=False` anyway).